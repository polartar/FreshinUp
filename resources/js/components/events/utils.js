import get from 'lodash/get'

const getFileNameCopy = (name) => {
  const regex = /\s*\(([0-9]+)\)$/gm
  const matches = name.match(regex) || []
  const count = (
    parseInt(
      get(matches, '[0]', '')
        .replace('(', '')
        .replace(')', '')
    ) || 0
  ) + 1
  return `Copy of ${name.replace(get(matches, '[0]', ''), '')} (${count})`
}

export default getFileNameCopy
