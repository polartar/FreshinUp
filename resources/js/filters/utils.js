export function prefixStr (value, prefix = '@') {
  if (!value) return ''
  return value ? `${prefix} ${value}` : ''
}
