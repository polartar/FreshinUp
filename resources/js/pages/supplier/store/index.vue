
<script>
import stores from '~/pages/admin/fleet-members/index.vue'
import { mapGetters } from 'vuex'
const INCLUDE = [
  'tags',
  'addresses',
  'status',
  'owner',
  'type'
]

export default {
  extends: stores,
  computed: {
    ...mapGetters(['currentUser'])
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.setPageLoading(true)
    Promise.all([
      vm.$store.dispatch('stores/setFilters', {
        ...vm.$route.query,
        ...this.lastFilterParams,
        owner_uuid: vm.currentUser.uuid
      }),
      vm.$store.dispatch('storeStatuses/getItems', {
        params: { include: INCLUDE }
      })
    ])
      .then(() => {
        if (next) next()
      })
      .catch((error) => console.error(error))
      .then(() => vm.setPageLoading(false))
  }
}
</script>
