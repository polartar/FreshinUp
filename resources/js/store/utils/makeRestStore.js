/**
 * All of this below is just so that when we call store.dispatch('module/deleteItem', { getItems: true })
 * it will get them with the needed include if that was set previously
 *
 * Example:
 * From
 * store.dispatch('module/getItems', { params: { include: 'submoduleA,submoduleB' } }
 * store.dispatch('module/deleteItem', { getItems: false, id: item.uuid })
 * store.dispatch('module/getItems', { params: { include: 'submoduleA,submoduleB' } }
 *
 * To
 * store.dispatch('module/getItems', { params: { include: 'submoduleA,submoduleB' } }
 * store.dispatch('module/deleteItem', { getItems: true, id: item.uuid })
 *
 * Hey that's one line less and a lot of code duplication avoided for each module
 */

import mapKeys from 'lodash/mapKeys'
import {
  _getPaginationAsParams, _getSortingAsParams,
  buildApi,
  makeModule as _makeModule
} from '@freshinup/core-ui/src/store/utils/makeRestStore'
import pickBy from 'lodash/pickBy'
import omitBy from 'lodash/omitBy'

export const makeModule = (store, moduleName = '') => {
  const m = _makeModule(store, moduleName)
  const _actions = store.actions

  m.mutations = {
    ...m.mutations,
    setFilters: (state, payload) => {
      payload = omitBy(payload, (value, key) => {
        return key.indexOf('sort') > -1
      })
      state.filters = mapKeys(payload, (value, key) => {
        if (['include'].includes(key)) {
          return key
        }
        return key.replace(/filter\[(.*)]/gi, '$1')
      })
    }
  }
  m.actions = {
    ...m.actions,
    getItems (context, payload = { params: {}, data: {} }) {
      const pagination = context.getters.pagination
      const sorting = context.getters.sorting
      // Convert filters into query parameters that is JSON API compliant
      const filters = mapKeys(context.state.filters, function (value, key) {
        // In order to be backwards compatible we need to ignore a few keys
        //    No deprecation plan yet as there is not a design change planned.
        //    Most likely we'll need a setSort() and term may remain
        if (['term', 'q', 'sort', 'include'].indexOf(key) > -1) {
          return key
        }
        return key.indexOf('filter[') === 0 ? key : `filter[${key}]`
      })

      // Deprecation Notice
      const deprecatedKeys = Object.keys(pickBy(payload.params, (value, key) => {
        return key.indexOf('filter[') > -1 || key.indexOf('sort') > -1
      }))
      if (deprecatedKeys.length > 0) {
        console.warn(`
FreshBUS: Version 1.15 of FreshBUS is the last support filtering through getList({ params }) on Store Modules made from makeRestStore.
Please switch to using setFilters() before calling getItems(). Keys provided are ${deprecatedKeys} on ${moduleName}
          `)
      }
      payload.params = {
        ..._getPaginationAsParams(pagination),
        ..._getSortingAsParams(sorting),
        ...filters,
        ...payload.params
      }

      return _actions.getItems.apply(null, [context, payload])
    }
  }
  return m
}

/**
 * Quickly build a REST Module with our Fresh Standards
 * @param name
 * @param items Initial State for Items
 * @param item  Initial State for Item (selected item)
 * @param filters
 * @param basePaths
 * @param options
 */
export default (name, { items, item, filters }, basePaths, options = {}) => {
  const api = buildApi(name, { items, item, filters }, basePaths, options)
  return makeModule(api.getStore(), name)
}
