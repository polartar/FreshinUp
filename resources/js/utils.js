import get from 'lodash/get'
/**
 *
 * @param options HTML attributes
 * @param [options.type] string 'link'|'script' or anything in fact
 * @param [options.container] HTMLElement
 * @returns {Promise<unknown>}
 */
export const lazyLoad = (options) => {
  return new Promise((resolve, reject) => {
    if (!options.type) {
      return reject(new Error('options.type is required'))
    }
    if (!options.container) {
      options.container = document.body
    }
    const { type, container, ...attributes } = options
    let element = document.createElement(type)
    Object.keys(attributes).forEach(key => element.setAttribute(key, attributes[key]))
    element.addEventListener('load', () => {
      resolve(element)
    })
    element.addEventListener('error', (e) => {
      reject(e)
    })
    container.append(element)
  })
}

export const getFileNameCopy = (name) => {
  const regex = /\s*\(([0-9]+)\)$/gm
  const matches = name.match(regex) || []
  const count = (
    parseInt(
      get(matches, '[0]', '')
        .replace('(', '')
        .replace(')', '')
    ) || 0
  ) + 1
  return `Copy of ${name.replace(get(matches, '[0]', ''), '')} (${count})`
}
