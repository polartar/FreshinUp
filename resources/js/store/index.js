import { createStore } from 'fresh-bus/store'

import page from './modules/page'
import devices from './modules/devices'
import financialModifiers from './modules/financialModifiers'
import financialReports from './modules/financialReports'
import paymentTypes from './modules/paymentTypes'
import financialsummary from './modules/financialsummary'
import squares from './modules/squares'
import documents from './modules/documents'
import documentStatuses from './modules/documentStatuses'
import documentTypes from './modules/documentTypes'
import transactions from './modules/transactions'
import companyDetails from './modules/companyDetails'
import eventSummary from './modules/eventSummary'
import eventTypes from './modules/eventTypes'
import eventHistories from './modules/eventHistories'
import venues from './modules/venues'
import stores from './modules/stores'
import storeStatuses from './modules/storeStatuses'
import storeTypes from './modules/storeTypes'
import companies from './modules/companies'

export default (initialState = {}) => {
  return createStore(
    initialState,
    {
      modules: {
        page: page({}),
        devices: devices({}),
        financialModifiers: financialModifiers({}),
        financialReports: financialReports({}),
        paymentTypes: paymentTypes({}),
        financialsummary: financialsummary({}),
        squares: squares({}),
        documents: documents({}),
        documentStatuses: documentStatuses({}),
        documentTypes: documentTypes({}),
        transactions: transactions({}),
        companyDetails: companyDetails({}),
        eventSummary: eventSummary({}),
        eventTypes: eventTypes({}),
        eventHistories: eventHistories({}),
        venues: venues({}),
        stores: stores({}),
        storeStatuses: storeStatuses({}),
        storeTypes: storeTypes({}),
        companies: companies({})
      }
    }
  )
}
