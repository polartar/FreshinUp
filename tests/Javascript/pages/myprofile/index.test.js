import { mount } from '@vue/test-utils'
import { createLocalVue } from 'tests/utils'
import { createStore } from 'fresh-bus/store'
import { FIXTURE_CURRENT_USER } from 'tests/__data__/user'
import Component from '~/pages/myprofile/index.vue'

describe('My profile page', () => {
  let localVue, mock, store
  beforeEach(() => {
    const vue = createLocalVue({ validation: true })
    localVue = vue.localVue
    mock = vue.mock
    mock
      .onGet('api/currentUser').reply(200, FIXTURE_CURRENT_USER)
      .onGet('api/users/1').reply(200, { data: FIXTURE_CURRENT_USER })
      .onAny().reply(config => {
        console.warn('No mock match for ' + config.url, config)
        return [404, {}]
      })
    store = createStore({
      currentUser: FIXTURE_CURRENT_USER
    })
  })
  afterEach(() => {
    mock.restore()
  })
  it('snapshot of page', async () => {
    const wrapper = mount(Component, {
      localVue: localVue,
      store
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
