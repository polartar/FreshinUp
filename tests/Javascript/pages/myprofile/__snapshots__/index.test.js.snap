// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My profile page snapshot of page 1`] = `<!---->`;
